### PR TITLE
support protobuf transformer

### PR DESCRIPTION
**Pull request by saitofun**

### CODE REVIEW

File: pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	if err = proto.Unmarshal(data, pv); err != nil {
+		return err
+	}
+	return nil
+}
+
+var (
+	ErrEncodeDataNotProtobuf   = errors.New("encode data must be `proto.Message`")
+	ErrDecodeTargetNotProtobuf = errors.New("decode target must be `proto.Message`")
+)
```
Comment (CoderZhi) [inline pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go:71]: move to top

File: pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	data, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+
+	if err = proto.Unmarshal(data, pv); err != nil {
+		return err
+	}
+	return nil
```
Comment (CoderZhi) [inline pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go:65]: `return proto.Unmarshal(data, pv)`

**Review by CoderZhi — APPROVED** [review]

File: pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go (hunk 1 above)
Comment (saitofun) [reply pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go]: Fixed

File: pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go (hunk 2 above)
Comment (saitofun) [reply pkg/depends/kit/httptransport/transformer/tsfm_protobuf.go]: Fixed

**Pull request merged**